### PR TITLE
enable conversation sessions by default

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- enable conversation sessions even if no `session_config` is given in the domain. This is a breaking change in contrast to Rasa Open Source 1 where this would result in disabled conversation sessions. The default value for `session_expiration_time` is the same one which we have been using in our `rasa init` project.
- update migration guide accordingly
- remove duplicate constants (caused when introducing `rasa.shared`).

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/migration-guide.mdx [hunk 1]
```diff
@@ -29,6 +29,17 @@ how you can migrate from one version to another.
   component in your `policies` configuration (`config.yml`) you can replace it
   with `TEDPolicy`. It accepts the same configuration parameters.
 
+* [Conversation sessions](domain.mdx#session-configuration) are now enabled by default
```
Comment (wochinge) [inline docs/docs/migration-guide.mdx:32]: The title of the targeted section is currently `Session Configuration`. I'd vote for renaming this to `Conversation Sessions`.

File: rasa/shared/core/domain.py [hunk 2]
```diff
@@ -185,7 +184,6 @@ def from_dict(cls, data: Dict) -> "Domain":
     def _get_session_config(session_config: Dict) -> SessionConfig:
         session_expiration_time_min = session_config.get(SESSION_EXPIRATION_TIME_KEY)
 
-        # TODO: 2.0 reconsider how to apply sessions to old projects and legacy trackers
```
Comment (erohmensing) [inline rasa/shared/core/domain.py:188]: I guess we've just decided to leave them as is?

**Review by erohmensing — APPROVED** [review]
Looks good, this behavior makes more sense with regards to the ["default configuration"](https://rasa.com/docs/rasa/domain#session-configuration) we already reference in the docs 👌

File: rasa/shared/core/domain.py (hunk 2 above)
Comment (wochinge) [reply rasa/shared/core/domain.py:188]: No, I changed the default from `0` to `60` which essentially enables them by default.

File: docs/docs/migration-guide.mdx (hunk 1 above)
Comment (wochinge) [reply docs/docs/migration-guide.mdx:32]: @erohmensing ?

Comment (erohmensing) [reply docs/docs/migration-guide.mdx:32]: Mm, I think it makes the most sense to have the title of the section to reflect the high level keys in the domain

Comment (wochinge) [reply docs/docs/migration-guide.mdx:32]: Okay, then let's leave it as it is.

File: rasa/shared/core/domain.py (hunk 2 above)
Comment (erohmensing) [reply rasa/shared/core/domain.py:188]: i guess i assumed that this note was about "should we read all of the conversations from 2018 as one conversation even if we introduce sessions now? with regards to applying them to legacy trackers

Comment (wochinge) [reply rasa/shared/core/domain.py:188]: Ah, okay. Yes, these will have sessions enabled, too, now

**Pull request merged**